jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelState.java

```diff
@@ -560,6 +564,11 @@ public void dispatch(ServletContext context, String path)
 
     protected void onTimeout()
     {
+        TimeoutException timeout = new TimeoutException("AsyncContext Timeout");
+        EndPoint endp = getHttpChannel().getEndPoint();
+        if (endp instanceof AbstractEndPoint)
+            ((AbstractEndPoint)endp).onIoFail(timeout);
```

Comment:
I think this is wrong, as we cannot do this for HTTP/2.
This is the high-layer for HTTP and we cannot shortcut and go straight to the `EndPoint`, we need to do different things depending on the transport.
We surely need to fail `HttpOutput` so that it can call `WriteListener.onError()`, but we cannot fail the whole `EndPoint` if a single request times out - not for HTTP/2.

Why cannot we do here just:

```java
_channel.abort(new TimeoutException());
```

Aborting the channel should do the right thing to input and output (i.e. fail them).